main/autogen/subclasses.cc

```diff
@@ -92,34 +84,57 @@ optional<Subclasses::SubclassInfo> Subclasses::descendantsOf(const Subclasses::M
 }
 
+const core::SymbolRef Subclasses::getConstantRef(const core::GlobalState &gs, string rawName) {
+    core::ClassOrModuleRef sym = core::Symbols::root();
+
+    for (auto &n : absl::StrSplit(rawName, "::")) {
+        const auto &nameRef = gs.lookupNameUTF8(n);
+        if (!nameRef.exists())
+            return core::Symbols::noSymbol();
+        sym = gs.lookupClassSymbol(sym, gs.lookupNameConstant(nameRef));
+    }
+    return sym;
+}
+
 // Manually patch the child map to account for inheritance that happens at runtime `self.included`
 // Please do not add to this list.
-void Subclasses::patchChildMap(Subclasses::Map &childMap) {
-    childMap["Opus::SafeMachine"].entries.insert(childMap["Opus::Risk::Model::Mixins::RiskSafeMachine"].entries.begin(),
-                                                 childMap["Opus::Risk::Model::Mixins::RiskSafeMachine"].entries.end());
+void Subclasses::patchChildMap(const core::GlobalState &gs, Subclasses::Map &childMap) {
+    auto safeMachineRef = getConstantRef(gs, "Opus::SafeMachine");
+    auto riskSafeMachineRef = getConstantRef(gs, "Opus::Risk::Model::Mixins::RiskSafeMachine");
+    if (!safeMachineRef.exists() || !riskSafeMachineRef.exists())
+        return;
+    childMap[safeMachineRef].entries.insert(childMap[riskSafeMachineRef].entries.begin(),
+                                            childMap[riskSafeMachineRef].entries.end());
 }
 
-vector<string> Subclasses::serializeSubclassMap(const Subclasses::Map &descendantsMap,
-                                                const vector<string> &parentNames) {
+vector<string> Subclasses::serializeSubclassMap(const core::GlobalState &gs, const Subclasses::Map &descendantsMap,
+                                                const vector<core::SymbolRef> &parentNames, const bool showPaths) {
     vector<string> descendantsMapSerialized;
+    const auto classFormatString = showPaths ? " class {} {}" : " class {}";
+    const auto moduleFormatString = showPaths ? " module {} {}" : " module {}";
 
-    for (const string &parentName : parentNames) {
-        auto fnd = descendantsMap.find(parentName);
+    for (const auto &parentRef : parentNames) {
+        auto fnd = descendantsMap.find(parentRef);
         if (fnd == descendantsMap.end()) {
             continue;
         }
         const Subclasses::SubclassInfo &children = fnd->second;
 
+        string parentName = parentRef.show(gs);
+
         auto type = children.classKind == ClassKind::Class ? "class" : "module";
         descendantsMapSerialized.emplace_back(fmt::format("{} {}", type, parentName));
 
         auto subclassesStart = descendantsMapSerialized.size();
-        for (const auto &[name, type] : children.entries) {
+        for (const auto &sym : children.entries) {
+            string_view path = gs.getPrintablePath(sym.loc(gs).file().data(gs).path());
+            string childName = sym.show(gs);
             // Ignore Modules
```

Comment:
Not sure why this comment existed.

Reply:
Yeah that doesn't make sense, seems to have been left in.